receiver/snmpreceiver/config.go

```diff
@@ -562,13 +571,38 @@ func validateResourceAttributeConfigs(cfg *Config) error {
 		return nil
 	}
 
-	// Make sure each Resource Attribute has either an OID or IndexedValuePrefix
+	// Make sure each Resource Attribute has exactly one of OID or ScalarOID or IndexedValuePrefix, and check that scalar and column OIDs end in the right digit 
 	for attrName, attrCfg := range resourceAttributes {
-		if attrCfg.OID == "" && attrCfg.IndexedValuePrefix == "" {
-			combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgResourceAttributeNoOIDOrPrefix, attrName))
+
+		hasOID := attrCfg.OID != ""
+		hasScalarOID := attrCfg.ScalarOID != ""
+		hasIVP := attrCfg.IndexedValuePrefix != ""
+
+		switch {
```

Comment:
here's an alternate for your switch. i "think" it's a little more concise/readable, but take it or leave it.

```
if hasOID {
  nums := strings.Split(attrCfg.OID, ".")
  if nums[len(nums) - 1] == "0" {
    combinedErr = multierr.Append(combinedErr, fmt.Errorf(errColumnOIDResourceAttributeEndsInZero, attrName, attrCfg.OID))
  }
}
if hasScalarOID {
  nums := strings.Split(attrCfg.ScalarOID, ".")
  if nums[len(nums) - 1] != "0" {
    combinedErr = multierr.Append(combinedErr, fmt.Errorf(errScalarOIDResourceAttributeEndsInNonzeroDigit, attrName, attrCfg.ScalarOID))
  }
}

noKeys := !(hasOID || hasScalarOID || hasIVP)
singleKey := hasOID ^ hasScalarOID ^ hasIVP

if noKeys {
  combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgResourceAttributeNoOIDOrScalarOIDOrPrefix, attrName))
  return combinedErr
}
if !singleKey {
  combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgMultipleKeysSetOnResourceAttribute, attrName))
}

return combinedErr
```

Reply:
I like it

Reply:
Actually, go has no logical xor so I'll keep as is.

Reply:
@kuiperda bummer. i just noticed that my code wasn't quite right even if ^ did work as a logical XOR.